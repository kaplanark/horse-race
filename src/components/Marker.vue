<script setup>
const props = defineProps({
	postion: {
		type: String,
		default: 'left'
	},
	text: {
		type: String,
		default: ''
	},
	count: {
		type: String,
		default: ''
	}
});
</script>

<template>
	<div class="marker" :class="'marker--' + props.postion">
		<span class="marker__text">{{ props.text }}</span>
		<span class="marker__count">{{props.count }}</span>
	</div>
</template>

<style lang="scss">
.marker {
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: center;

	&__text {
		font-size: 12px;
		color: var(--color-quaternary);
	}

	&__count {
		font-size: 12px;
		padding: 4px 8px;
		background-color: var(--color-quaternary);
		color: var(--color-white);
		border-radius: 12px;
		line-height: 1;
		position: relative;
	}

	&--right {
		align-items: flex-end;

		.marker__count {
			&::before {
				content: '';
				position: absolute;
				top: 50%;
				right: 100%;
				background-color: var(--color-quaternary);
				height: 1px;
				width: 100%;
			}
		}
	}

	&--left {
		align-items: flex-start;

		.marker__count {
			&::after {
				content: '';
				position: absolute;
				top: 50%;
				left: 100%;
				background-color: var(--color-quaternary);
				height: 1px;
				width: 100%;
			}
		}
	}
}
</style>