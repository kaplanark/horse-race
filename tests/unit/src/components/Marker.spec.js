import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import Marker from '@components/Marker.vue';

describe('Marker.vue', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(Marker, {
			props: {
				postion: 'left',
				text: 'Speed',
				count: '10km/sec',
			}
		});
	});

	afterEach(() => {
		wrapper.unmount();
	});

	test('is correctly rendered', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should renders is page content is correct', () => {
		expect(wrapper.vm).toBeTruthy();
	});

	test('renders props.postion when passed', () => {
		expect(wrapper.classes()).toContain('marker--left');
	});

	test('renders props.text when passed', () => {
		expect(wrapper.get('.marker__text').text()).toBe('Speed');
	});

	test('renders props.count when passed', () => {
		expect(wrapper.get('.marker__count').text()).toBe('10km/sec');
	});
});