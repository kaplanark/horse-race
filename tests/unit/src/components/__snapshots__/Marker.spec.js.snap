// Vitest Snapshot v1

exports[`Marker.vue > is correctly rendered 1`] = `"<div class=\\"marker marker--left\\"><span class=\\"marker__text\\">Speed</span><span class=\\"marker__count\\">10km/sec</span></div>"`;
